contracts/SmartOrderStrategy.sol

```diff
@@ -79,15 +84,20 @@ contract SmartOrderStrategy is ISmartOrderStrategy, AdminManagement {
             }
 
             // calculate input amount if ratio should be applied
-            if (_inputRatio != Constant.BPS_MAX) {
-                inputTokenBalance = (inputTokenBalance * _inputRatio) / Constant.BPS_MAX;
+            // we provide a _numerator and a _denominator to decide a ratio
+            if (_numerator != _denominator) {
+                inputTokenBalance = (inputTokenBalance * _numerator) / _denominator;
             }
             assembly {
                 mstore(add(_data, _dataOffset), inputTokenBalance)
             }
         }
 
-        (bool success, bytes memory result) = _dest.call{ value: _value }(_data);
+        bool success;
+        bytes memory result;
+        if (_value == 0) (success, result) = _dest.call(_data);
+        else (success, result) = _dest.call{ value: _value }(_data);
```

Comment:
What are the pros and cons of this change?

Reply:
I initially made this change to save gas, but after re-running it, I found that it actually consumes more gas. Maybe I used the wrong baseline for comparison before. I will remove this change